### PR TITLE
fix some typing issues

### PR DESCRIPTION
**Pull request by DanielIskandar**
*Start with a description of this PR. Then edit the list below to the items that make sense for your PR scope, and check off the boxes as you go!*

## Contributor Checklist

- [ ] I have broken down my PR scope into the following TODO tasks
   -  [ ] task 1
   -  [ ] task 2
- [ ] I have run the tests locally and they passed. (refer to testing section in [contributing](https://github.com/plotly/dash/blob/master/CONTRIBUTING.md))
- [ ] I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR

### optionals

- [ ] I have added entry in the `CHANGELOG.md`
- [ ] If this PR needs a follow-up in **dash docs**, **community thread**, I have mentioned the relevant URLS as follows
    -  [ ] this GitHub [#PR number]() updates the dash docs
    -  [ ] here is the show and tell thread in Plotly Dash community

### CODE REVIEW

File: dash/dash.py [hunk 1]
```diff
@@ -375,40 +378,43 @@ class Dash:
...
+        extra_hot_reload_paths: Optional[List[str]] = None,
+        plugins: Optional[List[Dict[str, Any]]] = None,
+        title: str = "Dash",
+        update_title: str = "Updating...",
+        long_callback_manager: Optional[
+            Any
+        ] = None,  # Type should be specified if possible
+        background_callback_manager: Optional[
+            Any
+        ] = None,  # Type should be specified if possible
```
Comment (T4rk1n) [inline dash/dash.py:413]: Both of those should be of type `dash.long_callback.managers.BaseLongCallbackManager`

File: dash/dash.py [hunk 2]
```diff
@@ -375,40 +378,43 @@ class Dash:
...
+        include_pages_meta: bool = True,
+        url_base_pathname: Optional[str] = None,
+        requests_pathname_prefix: Optional[str] = None,
+        routes_pathname_prefix: Optional[str] = None,
+        serve_locally: bool = True,
+        compress: Optional[bool] = None,
+        meta_tags: Optional[List[Dict[str, Any]]] = None,
+        index_string: str = _default_index,
+        external_scripts: Optional[List[Union[str, Dict[str, Any]]]] = None,
+        external_stylesheets: Optional[List[Union[str, Dict[str, Any]]]] = None,
```
Comment (T4rk1n) [inline dash/dash.py:400]: Those `Any` in the `Dict` would be `str`

File: dash/dash.py [hunk 3]
```diff
@@ -375,40 +378,43 @@ class Dash:
     STARTUP_ROUTES: list = []
```
Comment (T4rk1n) [inline dash/dash.py:379]: we can add `server: flask.Flask` here.

**Review by T4rk1n — COMMENTED** [review]
This look good overall and a good addition, just need to address my previous comments and add a changelog entry.

**Comment by gvwilson** [comment]
@DanielIskandar are you interested in updating this one so that we can merge it? thanks - @gvwilson

**Review by T4rk1n — APPROVED** [review]

**Pull request merged**